test/extended/images/mysql_replica.go

```diff
@@ -0,0 +1,143 @@
+package images
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/rand"
+)
+
+var _ = g.Describe("images: MySQL replication template", func() {
+	defer g.GinkgoRecover()
+	var (
+		persistentVolumes []*kapi.PersistentVolume
+		templatePath      = "https://raw.githubusercontent.com/openshift/mysql/master/5.5/examples/replica/mysql_replica.json"
+		oc                = exutil.NewCLI("mysql-replication", exutil.KubeConfigPath())
+		user              = "user"
+		userPass          = "userpass"
+		rootPass          = "rootpass"
+		database          = "userdb"
+	)
+
+	g.Describe("MySQL replication", func() {
+		g.JustBeforeEach(func() {
+			var err error
+			persistentVolumes, err = exutil.SetupHostPathVolumes(oc.AdminKubeREST().PersistentVolumes(), "mysql-replication", "512Mi", 1)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.AfterEach(func() {
+			err := oc.AsAdmin().Run("delete").Args("all", "--all", "-n", oc.Namespace()).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			err = oc.AsAdmin().Run("delete").Args("pvc", "--all", "-n", oc.Namespace()).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			err = exutil.CleanupHostPathVolumes(oc.AdminKubeREST().PersistentVolumes(), "mysql-replication")
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		// TODO: Add parallel loop for 5.6 templates
+		g.It("should replicate data", func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			g.By(fmt.Sprintf("processing and creating the template %q", templatePath))
+			configFile, err := oc.Run("process").
+				Args(
+				"-f", templatePath,
+				"-v", "MYSQL_USER="+user+",MYSQL_PASSWORD="+userPass+",MYSQL_ROOT_PASSWORD="+rootPass+",MYSQL_DATABASE="+database).
+				OutputToFile("config.json")
```

Comment:
You can create the template in OpenShift and then instantiate it without need of storing temporary file (I hope). Can you check if that approach is doable?